models/pokemon.js

```diff
@@ -287,9 +287,12 @@ Pokemon.prototype.assignMoves = function () {
     this.pokemonData.moves.forEach(nextMove =>
     {
         let newestAppearance = nextMove.version_group_details.length - 1;
+        let secondNewestAppearance = nextMove.version_group_details.length -2;
         let learnMethod = nextMove.version_group_details[newestAppearance].move_learn_method.name;
         let levelLearned = nextMove.version_group_details[newestAppearance].level_learned_at;
-        if (learnMethod === "level-up" && levelLearned <= vgLevel) { legalMoves.push(nextMove) }
+        let secondLearnMethod = nextMove.version_group_details[secondNewestAppearance].move_learn_method.name;
+        let secondLevelLearned = nextMove.version_group_details[secondNewestAppearance].level_learned_at;
+        if ((learnMethod === "level-up" && levelLearned <= vgLevel) || (secondLearnMethod === "level-up" && secondLevelLearned <= vgLevel)) { legalMoves.push(nextMove) }
```

Comment:
So this is mostly doubled up for moves that have an appearance as a TM & a learned move, since (as far as I've seen) if they have it available as both the TM version is the one that shows up second, meaning that the random move selection won't pick that move despite it being a learned-by-level.

It also shoooouldn't be an issue in regards to stuff like Magikarp, since version_group_details is a list of every *gen* the pokemon could learn it. However, I do need to do that for handling pokemon moves that are *only* learned via one way from SuMo onward- right now it's guaranteed not to break because Ultra makes it so every move has at least two versions, but when they eventually add SwSh moves it'll break. Gonna fix that today I think

Reply:
Moved this into a function and proofed it for if there's only one item in version_group_details